### PR TITLE
Confidentiality

### PR DESCRIPTION
**Pull request by cfcs**
ping @sshine 

Here are the patches for `passive` confidentiality queries, and a bunch of other misc fixes and tests.

Also implemented is generation of PROLOG syntax from the Verifpal models and `dot` graph generation.

### CODE REVIEW

File: src/VerifPal/Pretty.hs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    ++ ( map constantName consts
+           & Text.intercalate ", "
+           & Text.unpack
+       )
+prettifyQuery k = show k
+
+-- see: https://github.com/Mesabloo/diagnose/blob/master/test/rendering/Spec.hs
+instance HasHints Void Text where
+  -- hints :: Void -> [Text]
+  hints _ = ["TODOwhatisthis"]
```
Comment (sshine) [inline src/VerifPal/Pretty.hs:30]: I'm not sure if this type class instance is required by something?

**Review by sshine — APPROVED** [review]
I've moved PropertyTest.hs.bak into test/ and made it compatible with [tasty-discover](https://github.com/haskell-works/tasty-discover#write-tests), meaning `stack test` will scan through the test suite for top-level identifiers like `hprop_` and run them. That is, the `main` of the test suite is implicitly generated.

I made a few other minor changes.

The things that seem partially done will be merged as-is, we can continue when convenient.

File: src/VerifPal/Pretty.hs (hunk 1 above)
Comment (cfcs) [reply src/VerifPal/Pretty.hs:30]: It was required by the `diagnose` parser library; or at least it was complaining without this. I don't really understand *why*, but it didn't seem to build without it.

**Comment by cfcs** [comment]
let's get it merged @sshine?

**Pull request merged**